google/cloud/bigquery/job.py

```diff
@@ -3275,6 +3283,27 @@ def result(
         rows._preserve_order = _contains_order_by(self.query)
         return rows
 
+    def _get_progress_bar(self, progress_bar_type, description, total, unit):
```

Comment:
Seems like this is very close to the logic in `table.py`

https://github.com/googleapis/python-bigquery/blob/20f473bfff5ae98377f5d9cdf18bfe5554d86ff4/google/cloud/bigquery/table.py#L1374

I'd suggest creating a `_tqdm_helpers.py` module similar to our pandas helpers to hold this logic for both Table and Job logic.